starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaCodecBridge.java

```diff
@@ -957,45 +957,43 @@ private int queueSecureInputBuffer(
       int[] numBytesOfEncryptedData,
       int numSubSamples,
       int cipherMode,
-      int blocksToEncrypt,
-      int blocksToSkip,
+      int patternEncrypt,
+      int patternSkip,
       long presentationTimeUs) {
     resetLastPresentationTimeIfNeeded(presentationTimeUs);
     try {
+      boolean usesCbcs = cipherMode == MediaCodec.CRYPTO_MODE_AES_CBC;
+      if (usesCbcs && !MediaDrmBridge.isCbcsSchemeSupported()) {
+        Log.e(TAG, "Encryption scheme 'cbcs' not supported on this platform.");
+        return MediaCodecStatus.ERROR;
+      }
       CryptoInfo cryptoInfo = new CryptoInfo();
       cryptoInfo.set(
           numSubSamples, numBytesOfClearData, numBytesOfEncryptedData, keyId, iv, cipherMode);
-
-      if (cipherMode == MediaCodec.CRYPTO_MODE_AES_CBC) {
-        cryptoInfo.setPattern(new Pattern(blocksToEncrypt, blocksToSkip));
-      } else if (blocksToEncrypt != 0 || blocksToSkip != 0) {
-        Log.e(TAG, "Pattern encryption only supported for 'cbcs' scheme (CBC mode).");
-        return MediaCodecStatus.ERROR;
+      if (patternEncrypt != 0 && patternSkip != 0) {
+        if (usesCbcs) {
+          // Above platform check ensured that setting the pattern is indeed supported.
+          cryptoInfo.setPattern(new Pattern(patternEncrypt, patternSkip));
+        } else {
+          Log.e(TAG, "Pattern encryption only supported for 'cbcs' scheme (CBC mode).");
+          return MediaCodecStatus.ERROR;
+        }
```

Comment:
Just wanted to double check that we have verified the new implementation against cbcs and cenc playbacks.

Reply:
Yes, cenc playbacks live on YouTube, cbcs and cenc streams from [go/cobalt-media-features#encrypted-content](go/cobalt-media-features#encrypted-content)